### DISCUSSION TITLE
Is it possible to configure settings per endpoint?

### DISCUSSION THREAD

**Discussion by conioh** [neutrinolabs/xrdp#3049] [Q&A] [answered]
The issue was raised in #1124:

> With the added support for `AF_VSOCK` some users would like to be able to connect both hypervisor local and also expose a remote endpoint at the same time. This would be a modification to the config to support both `tcp` settings and `vsock` settings for the same daemon at the same time. Then a user could connect via a fast connection (ip independent) when local but also connect to this vm when remote. This is roughly exactly what the windows `TermSrv` does to accomplish this.
> 
> A few thoughts:
> 
> * We would need to modify the config to support connection independent settings and add new connection dependent settings sections. IE: you might want `bitmap_compression=true` for `tcp` but `=false` for `vsock`.
> 
> ...

Soon after [the following comment](https://github.com/neutrinolabs/xrdp/issues/1124#issuecomment-507480818) the issue was closed:

> I think all the items are addressed and I created PR #1366

But as far as I can see that PR only added support for multiple endpoints, but not for settings per endpoint. Is there a way to configure settings per endpoint?

For example, I'd like the equivalent of:
```
[Globals]
port=vsock://-1:3389 tcp://:3389
security_layer=negotiate
crypt_level=high
bitmap_compression=true

[Globals_vsock://-1:3389]
security_layer=rdp
crypt_level=none
bitmap_compression=false
```

That is, to have some `[Globals]` for the general case and endpoint-specific `[Globals]` for one the multiple endpoint specified by the `port` parameter (in this case, for Hyper-V Enhanced Session Mode).

**Comment by matt335672** [comment, marked as answer]
No, this isn't currently possible. At the moment, only separate endpoints are supported.

At the same time, it's an interesting idea so I'll suggest it as an enhancement.
